types/chroma-js/chroma-js-tests.ts

```diff
@@ -187,3 +190,9 @@ function test_types() {
     const color: chroma.Color = chroma('orange');
     const scale: chroma.Scale = chroma.scale('RdYlBu');
 }
+
+// the following should actually, pass, but TS can't disambiguate between a parameter
+// which is passed as undefined/null or not passed at all
+// const scaleColors1: Color[] = chroma.scale(['black', 'white']).colors(12);
+const scaleColors2: Color[] = chroma.scale(['black', 'white']).colors(12, null);
+const scaleColors3: Color[] = chroma.scale(['black', 'white']).colors(12, undefined);
```

Comment:
The bug was that, as the format parameter was marked as optional for all overloads, TS would pick the first one when null or undefined was passed, resulting in string[] cannot be assigned to Color[]